### DISCUSSION TITLE
Is linked_places always (for cities, towns and villages relations) the node representing the city (or at list the point where the name is displayed on the map) ?

### DISCUSSION THREAD

**Discussion by Sliverside** [osm-search/Nominatim#3337] [Q&A] [answered]
Hi, I woundering what is a linked place ? the only thing that i found online is this page : [Details - Nominatim 4.3.2 Manual](https://nominatim.org/release-docs/latest/api/Details/#output-details), but I not sure to understand it well.

I have a database with user-given city name and zipcode (in France and Swizerland) and I asociated them with an OSM relation. I want to associate them to an OSM node also and so I look at the members of the relation and I take the one with the "admin_centre" role but many of them (almost only swiss ones) doesn't have it.

Using Nominatim, some of this cities have a linked_place, is it always (for cities, towns and villages) the node representing the city (or at list the point where the name is displayed on the map) ?

Exemple of relations with a linked place but no admin center :
- [Plan-les-Ouates](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1685523)
- [Coppet](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1684917)
- [Isérables](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1685319)
- [Versoix](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1685541)
- [Confignon](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1685472)

Bonus question: linked_places seems to be unidirectional (a relation is linked to a node but the node isn't linked to it), why is it like that ?
Bonus question 2: in some cities (exemple: [Crans-Montana](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=6820012)) there is no "admin_centre" and no "linked place", what is the little circle on the map ?

Thanks !

**Comment by lonvia** [comment, marked as answer]
The linked place is the place node that most likely represents the same place as the boundary relation. So it is kind of what you are looking for. Nominatim uses some heuristics to determine the place. They use the wikidata tag, name, place type. If you are interested in the details, the code is [here](https://github.com/osm-search/Nominatim/blob/master/lib-sql/functions/placex_triggers.sql#L256). The `admin_center` role is not really suitable as it usually points to the capital of the boundary which is not always the same thing. The `label` role is a more reliable candidate but also used relatively rarely.

> Bonus question: linked_places seems to be unidirectional (a relation is linked to a node but the node isn't linked to it), why is it like that ?

There is an index on the column that allows a lookup in the opposite direction. No need to duplicate the information.

> Bonus question 2: in some cities (exemple: [Crans-Montana](https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=6820012)) there is no "admin_centre" and no "linked place", what is the little circle on the map ?

It is simply the geometric center of the boundary.

**Reply by Sliverside** [reply]
Thanks a lot for your time. This is really helpful !
